File: CodingInterviews/ReplaceSpace.java
class Solution {
    public String replaceSpace(String s) {
        StringBuilder ans = new StringBuilder();
        for (char c : s.toCharArray()) {
            if (c == ' ')
                ans.append("%20");
            else
                ans.append(c);
        }
        return ans.toString();
    }
}

public class ReplaceSpace {
    public static void main(String[] args) {
        String s = "We are happy.";
        var sol = new Solution();
        System.out.println(sol.replaceSpace(s));
    }
}

File: NowCoder/CodingInterviews/ReplaceSpace.cpp
#include "main.h"

class Solution {
  public:
    void replaceSpace(char *str, int length) {
        for (int i = 0; i < length; i++)
            if (str[i] == ' ') {
                for (int j = length + 2; j > i; --j)
                    str[j] = str[j - 2];
                length += 2;
                str[i] = '%';
                str[i + 1] = '2';
                str[i + 2] = '0';
                i += 2;
            }
    }
};